Realm/RLMConfiguration.m

```diff
@@ -0,0 +1,159 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMConfiguration_Private.h"
+
+static NSString * const c_RLMConfigurationProperties[] = {
+    @"path",
+    @"inMemoryIdentifier",
+    @"encryptionKey",
+    @"readOnly",
+    @"schemaVersion",
+    @"migrationBlock",
+    @"dynamic",
+    @"customSchema",
+};
+static const NSUInteger c_RLMConfigurationPropertiesCount = sizeof(c_RLMConfigurationProperties) / sizeof(NSString *);
+
+@interface RLMConfiguration ()
+
+@property (nonatomic, copy, readwrite) NSString *path;
+@property (nonatomic, copy, readwrite) NSString *inMemoryIdentifier;
+@property (nonatomic, copy, readwrite) NSData *encryptionKey;
+@property (nonatomic, readwrite)       BOOL    readOnly;
+@property (nonatomic, copy, readwrite) NSString *fileProtectionAttributes;
+
+@property (nonatomic, readwrite)       NSUInteger schemaVersion;
+@property (nonatomic, copy, readwrite) RLMMigrationBlock migrationBlock;
+
+@end
+
+@interface RLMConfiguration ()<RLMConfigurator>
```

Comment:
Why the empty interface? You could move the RLMConfigurator conformance to the previous interface in this file.